src/nvim/normal.c

```diff
@@ -2790,7 +2790,7 @@ void pop_showcmd(void)
 
 static void display_showcmd(void)
 {
-  if (p_ch < 1 && !ui_has(kUIMessages)) {
+  if (!ui_has_messages()) {
```

Comment:
showcmd and ruler definitely wouldn't be shown with cmdheight=0, so this kind of guard is correct.

Reply:
What is the problem?

Reply:
This is correct. There is no problem here.

Reply:
If I remove the guard, it is failed by `grid.c:335: grid_puts_line_flush: Assertion `put_dirty_row != -1' failed.`.